grep issues running on Mac OSX 13.2 (Ventura)

Tried to use this action on a local Github Action runner on a Mac Studio M1 and it breaks on a grep syntax issue:

```
Run export LC_ALL=en_US.utf8
  export LC_ALL=en_US.utf8
  git log $(git describe --abbrev=0 2> /dev/null || git rev-list --max-parents=0 HEAD)..HEAD | \
    grep -oP "DNA-\d{1,}" | sort | uniq | \
    sed 's/^\|$/"/g' | paste -sd , | awk '{print "RELATED_JIRA_ISSUES="$0}' >> $GITHUB_ENV
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
usage: paste [-s] [-d delimiters] file ...
Error: Process completed with exit code 1.
```

`grep --version` on the machine gives me `grep (BSD grep, GNU compatible) 2.6.0-FreeBSD`

Apparently the `-P` option is not available on this version of grep.

On this man page[ubuntu grep](https://manpages.ubuntu.com/manpages/bionic/en/man1/grep.1.html) it states that -P is experimental, so maybe good idea to use some other regex syntax?

--- a/action.yml
+++ b/action.yml
@@ -21,7 +21,7 @@ runs:
       run: |
         export LC_ALL=en_US.utf8
         git log $(git describe --abbrev=0 2> /dev/null || git rev-list --max-parents=0 HEAD)..HEAD | \
-          grep -oP "${{ inputs.jira-project-key }}-\d{1,}" | sort | uniq | \
+          grep -oE "${{ inputs.jira-project-key }}-[[:digit:]]{1,}" | sort | uniq | \
           sed 's/^\|$/"/g' | paste -sd , | awk '{print "RELATED_JIRA_ISSUES="$0}' >> $GITHUB_ENV
       shell: bash
 

